src/formatDate.js

```diff
@@ -8,7 +9,41 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separador = fromFormat[3];
+  const separadorTo = toFormat[3];
+
+  const newDate = date.split(separador);
+
+  const dateObj = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    dateObj[fromFormat[i]] = newDate[i];
+  }
+
+  const newDateParts = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const formatKey = toFormat[i];
+
+    if (formatKey === 'YYYY' && dateObj['YY']) {
+      // Converter YY para YYYY
+      const year = parseInt(dateObj['YY'], 10);
+
+      dateObj['YYYY'] = year < 30 ? '20' + dateObj['YY'] : '19' + dateObj['YY'];
+    }
+
+    if (formatKey === 'YY' && dateObj['YYYY']) {
+      // Converter YYYY para YY
+      dateObj['YY'] = dateObj['YYYY'].slice(-2);
+    }
+
+    newDateParts.push(dateObj[formatKey]);
```

Comment:
The second for loop iterates over `toFormat` and assumes that `fromFormat` and `toFormat` will always contain the same date parts, just possibly in a different order and with different separators. This logic works under the assumption that the input is always correct, as stated in the task description. However, it might be beneficial to add a comment explaining this assumption for clarity and future maintenance.